recipes/openssl/3.x.x/conanfile.py

```diff
@@ -0,0 +1,630 @@
+    settings = "os", "compiler", "arch", "build_type"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/openssl/openssl"
+    license = "Apache-2.0"
+    topics = ("conan", "openssl", "ssl", "tls", "encryption", "security")
+    description = "A toolkit for the Transport Layer Security (TLS) and Secure Sockets Layer (SSL) protocols"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "386": [True, False],
+        "capieng_dialog": [True, False],
+        "enable_capieng": [True, False],
+        "no_asm": [True, False],
+        "no_async": [True, False],
+        "no_bf": [True, False],
+        "no_cast": [True, False],
+        "no_deprected": [True, False],
+        "no_des": [True, False],
+        "no_dh": [True, False],
+        "no_dsa": [True, False],
+        "no_dso": [True, False],
+        "no_engine": [True, False],
+        "no_fips": [True, False],
+        "no_hmac": [True, False],
+        "no_legacy": [True, False],
+        "no_md5": [True, False],
+        "no_mdc2": [True, False],
+        "no_rc2": [True, False],
+        "no_rc4": [True, False],
+        "no_rsa": [True, False],
+        "no_sha": [True, False],
+        "no_sse2": [True, False],
+        "no_threads": [True, False],
+        "openssldir": "ANY",
+        "zlib": [True, False]
+        }
+    default_options = {key: False for key in options.keys()}
+    default_options["fPIC"] = True
+    default_options["zlib"] = True
+    default_options["openssldir"] = None
+    _env_build = None
+    _source_subfolder = "source_subfolder"
+    exports_sources = ["patches/*"]
+
+    def build_requirements(self):
+        if tools.os_info.is_windows:
+            if not self._win_bash:
+                self.build_requires("strawberryperl/5.30.0.1")
+            if not self.options.no_asm and not tools.which("nasm"):
+                self.build_requires("nasm/2.14")
+        if self._win_bash:
+            if "CONAN_BASH_PATH" not in os.environ:
+                self.build_requires("msys2/20200517")
+
+    def requirements(self):
+        if self.options.get_safe("zlib") == True:
+            self.requires("zlib/1.2.11")
+
+    @property
+    def _is_msvc(self):
+        return self.settings.compiler == "Visual Studio"
+
+    @property
+    def _is_clangcl(self):
+        return self.settings.compiler == "clang" and self.settings.os == "Windows"
+
+    @property
+    def _is_mingw(self):
+        return self.settings.os == "Windows" and self.settings.compiler == "gcc"
+
+    @property
+    def _use_nmake(self):
+        return self._is_clangcl or self._is_msvc
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def config_options(self):
+        if self.settings.os != "Windows":
+            del self.options.capieng_dialog
+            del self.options.enable_capieng
+        else:
+            del self.options.fPIC
```

Comment:
swap `config_options` and `configure` + remove `fPIC` when shared.

```suggestion
    def config_options(self):
        if self.settings.os != "Windows":
            del self.options.capieng_dialog
            del self.options.enable_capieng
        else:
            del self.options.fPIC

    def configure(self):
        if self.options.shared:
            del self.options.fPIC
        del self.settings.compiler.libcxx
        del self.settings.compiler.cppstd
```